R/html_document_base.R

```diff
@@ -89,10 +89,16 @@ html_document_base <- function(theme = NULL,
     format_deps <- append(format_deps, html_dependency_header_attrs())
     if (!is.null(theme)) {
       format_deps <- append(format_deps, list(html_dependency_jquery()))
-      format_deps <- append(format_deps, bootstrap_dependencies(
-        # If TRUE, an as_bs_theme(theme) has been set globally (so users may customize it)
-        if (is_bs_theme(theme)) bslib::bs_global_get() else theme
-      ))
+      # theme was set globally pre-knit, and it may be modified during knit
+      if (is_bs_theme(theme)) {
+        theme <- bslib::bs_global_get()
+      }
+      bootstrap_deps <- if (is_bs_theme(theme) && is_shiny(runtime)) {
+        list(shiny_bootstrap_lib(theme))
+      } else {
+        bootstrap_dependencies(theme)
+      }
+      format_deps <- append(format_deps, htmltools::resolveDependencies(bootstrap_deps))
```

Comment:
@cpsievert just curious why do we need `htmltools::resolveDependencies` exactly ?
It is for my understanding.

Reply:
Is this in case someone modifies the theme from inside the document ?

Reply:
It's because `shiny::bootstrapLib()` returns a `htmltools::tagFunction()` which needs `htmltools::resolveDependencies()` to be called on it in order to generate the dependency. This is really just be being paranoid about a `dependency_resolver` being specified that doesn't call `htmltools::resolveDependencies()`

Reply:
OK great thanks! still learning all this details. 👍